### DISCUSSION TITLE
How can I prevent duplicate fetch

### DISCUSSION THREAD

**Discussion by WreewanMorhee** [remix-run/remix#7766] [Q&A]
I know `shouldRevalidate` this function
but let me describe my question first:

I go to first page: mydomain/l?page=1  (fetch data about page=1, we assume data here called A_Data)
and got to next page : mydomain/l?page=2  (fetch data about page=2, we assume data here called B_Data)
then 
when I back to mydomain/l?page=1 
How should I still using previous A_Data and prevent duplicate fetching about page=1 ?

**Comment by sergiodxa** [comment]
You need to cache it, one way could be to use an HTTP cache with the Cache-Control header to cache the whole response, note that the document response and the data response will have separate caches in this case.

Another way is to use a server-side cache (e.g. lru-cache or Redis) to cache the data server-side, which will let you have more control and even share the cache between users if you want.

https://sergiodxa.com/articles/http-vs-server-side-cache-in-remix

**Comment by suhaotian** [comment]
Check this: https://github.com/suhaotian/xior?tab=readme-ov-file#tips-make-your-ssrserver-side-rendering-app-more-stable-and-faster
